pkg/apiserver/resthandler.go

```diff
@@ -150,30 +147,15 @@ func getRequestOptions(req *restful.Request, scope RequestScope, kind string, su
 		query = newQuery
 	}
 
-	// TODO Options a mess.  Basically the intent is:
-	// 1. try to decode using the expected external GroupVersion
-	// 2. if that fails, fall back to the old external serialization being used before, which was
-	//    "v1" and decode into the unversioned/legacykube group
-	gvString := scope.APIVersion
-	internalGVString := scope.InternalVersion.String()
-
-	versioned, err := scope.Creater.New(gvString, kind)
+	versioned, err := scope.Creater.New(externalFQKind.GroupVersion().String(), externalFQKind.Kind)
 	if err != nil {
-		gvString = "v1"
```

Comment:
> Why is this hack not needed anymore? IIUC, we need this hack because the unversioned options are not register in all groups. Has someone fixed it? I know wojtek-t has a PR doing that, but it's not merged yet.

The `externalFQKind` says exactly what this method should attempt to decode, so it now has enough information to get the decoding right on the first try.